### PR TITLE
Update BUILD.md

### LINKED ISSUE #5063: BUILD.md build architecture outdated
### What problem are you trying to solve?
Looking at the BUILD.md file it seems that the Build Architecture graph is outdated. The graph originates from one single workflow.yml file but in reality several files exists.

### How should the problem be solved?

Update the graph to properly display what happens in the build workflow.

### Any alternatives you've considered?

N/A

### How would users interact with this feature?

Gain knowledge about how the build architecture actually works, helping further development.

### PR DESCRIPTION
**Pull request by alpeb**
Fixes #5063

- Completed table of contents.
- Added references to viz and other extensions there where appropriate.
- Updated components chart.
- Updated build architecture chart.
- Moved `go test` paragraphs into `TEST.md`, and moved from `TEST.md` the linting paragraph.
- Group Helm paragraphs together.
- Deleted the "Go modules and dependencies" paragraph. We can assume most people are already familiar with go modules.

### CODE REVIEW

**Comment by alpeb** [comment]
This is how the updated Components chart looks like:
image

OTOH the site to generate the build architecture seems down...

**Review by dadjeibaah — APPROVED** [review]
LGTM! 📦 🚢

**Comment by alpeb** [comment]
You can check the new dependencies graph here:
https://g.gravizo.com/source/svg/build_architecture?https%3A%2F%2Fraw.githubusercontent.com%2Flinkerd%2Flinkerd2%2F8705367329891ef363a8a909cdba612c9d2c4aca%2FBUILD.md
So big as to not be useful anymore?

**Comment by alpeb** [comment]
Also, after the linkerd-await change we're no longer supporting plain vanilla docker and instead require buildx always, so I updated the doc accordingly.

**Comment by alpeb** [comment]
After some discussion, we decided to remove the bulid dependencies graph.

File: BUILD.md [hunk 1]
```diff
@@ -146,33 +176,32 @@ curl https://run.linkerd.io/emojivoto.yml | bin/linkerd inject - | kubectl apply
...
 
 Control Plane components have the `trace-collector` flag used to enable
 [Distributed Tracing](https://opentracing.io/docs/overview/what-is-tracing/) for
 development purposes. It can be enabled globally i.e Control plane components
-and their proxies by using the `--control-plane-tracing` installation flag.
+and their proxies by using the `--set controlPlaneTracing=true` installation
+flag.
 
 This will configure all the components to send the traces at
 `linkerd-collector.{{.Namespace}}.svc.{{.ClusterDomain}}:55678`
```
Comment (adleong) [inline BUILD.md:194]: Should this be `collector.{{.Values.controlPlaneTracingNamespace}}.svc.{{.Values.clusterDomain}}:55678`?

File: BUILD.md [hunk 2]
```diff
@@ -146,33 +176,32 @@ curl https://run.linkerd.io/emojivoto.yml | bin/linkerd inject - | kubectl apply
...
 
 # install Linkerd with tracing
-linkerd install --control-plane-tracing | kubectl apply -f -
+linkerd install --set controlPlaneTracing=true | kubectl apply -f -
 
-# install OpenCensus collector and Jaeger collector to collect traces
-linkerd inject https://gist.githubusercontent.com/Pothulapati/245842ce7f319e8bcd02521460684d6f/raw/52c869c58b07b17caeed520aa91380c2230d6e0c/linkerd-tracing.yaml --manual | kubectl apply -f -
+# install the Jaeger extension
+linkerd jaeger install | kubectl apply -f -
 ```
```
Comment (adleong) [inline BUILD.md:203]: perhaps add a note that control plane components need to be restarted after the jaeger-injector is deployed before they will be able to export traces.

File: BUILD.md [hunk 3]
```diff
@@ -517,96 +488,42 @@ Whenever you make changes to the files under
...
-
-    "docker-push" -> "_docker.sh";
-
-    "docker-retag-all" -> "_docker.sh";
-
-    "go-run" -> ".gorun";
-    "go-run" -> "root-tag";
-
-    "linkerd" -> "build-cli-bin";
+### Generating helm charts docs
```
Comment (kleimkuhler) [inline BUILD.md:491]:
```suggestion
### Generating Helm charts docs
```

**Review by kleimkuhler — COMMENTED** [review]
In addition to what Alex has mentioned already just a small nit

File: BUILD.md (hunk 1 above)
Comment (alpeb) [reply BUILD.md:194]: :+1:

File: BUILD.md (hunk 2 above)
Comment (alpeb) [reply BUILD.md:203]: :+1:

File: BUILD.md (hunk 3 above)
Comment (alpeb) [reply BUILD.md]: :+1:

**Review by kleimkuhler — APPROVED** [review]
👍

**Pull request merged**